mobile/components/views/common/AttractionDetail.tsx

```diff
@@ -1,76 +1,120 @@
+  const [reviewsModalVisible, setReviewsModalVisible] = useState<boolean>(false);
+  const userId = 1; //ToDo it's temporary
 
+  useEffect(() => {
+    const attractionId = route.params.id;
+    fetchAttraction(attractionId)
+      .then((attraction) => setAttractionData(attraction))
+      .catch(() => {
+        setAttractionData(attractionsExamples.find((attraction) => attraction.id === attractionId));
+        Alert.alert('Error', 'Failed getting attraction. Loading attraction demo data');
+      });
+    //Add another fetches
+    setAttractionImages(
+      Array.from({ length: 3 }, (_, index) => (
+        <View key={index} style={imagePlaceholder.placeholder} />
+      ))
+    );
+  }, [route.params.id]);
 
-    useEffect(() => {
-        const attractionId = route.params.id;
-        fetchAttraction(attractionId)
-            .then((attraction) => setAttractionData(attraction))
-            .catch((error) => {
-                setAttractionData(attractionsExamples.find(attraction => attraction.id === attractionId));
-                Alert.alert('Error', 'Failed getting attraction. Loading attraction demo data');
-            });
-        //Add another fetches
-        setAttractionImages(Array.from({ length: 3 }, (_, index) => (
-            <View key={index} style={imagePlaceholder.placeholder} />
-        )));
-    }, []);
+  const editBox = (attractionAuthor: number | undefined) => {
+    if (attractionAuthor && attractionAuthor === userId) {
+      return (
+        <TouchableOpacity onPress={() => setEditMenuVisible(true)}>
+          <Text style={attractionDetails.name}>✎</Text>
+        </TouchableOpacity>
+      );
+    }
+  };
 
-    return (
-        <>
-            <ScrollView style={attractionDetails.container}>
-                <View style={attractionDetails.header}>
-                    <Text style={attractionDetails.name}>{attractionData?.name}</Text>
-                    <TouchableOpacity style={attractionDetails.ratingContainer}>
-                        <View style={attractionDetails.ratingBox}>
-                            <Text style={attractionDetails.ratingText}>
-                                {attractionData?.rating.toFixed(1)}
-                            </Text>
-                            <View style={attractionDetails.addOpinionIcon}>
-                                <Text style={attractionDetails.addOpinionText}>+</Text>
-                            </View>
-                        </View>
-                    </TouchableOpacity>
-                </View>
+  const onGeneralRatingClick = () => {
+    setReviewsModalVisible(true);
+  };
 
-                <View style={attractionDetails.categoryContainer}>
-                    <Text style={attractionDetails.category}>{attractionData?.category}</Text>
-                    <Text style={attractionDetails.views}>[100 views]</Text>
-                </View>
+  return attractionDetails ? (
+    <>
```

Comment:
Do we really need to wrap it with <> </> ? Im not sure but if we need this it can stay

Reply:
Yes because only one parent element needs to be returned and react element is the lightest.